src/mlpack/methods/ann/models/yolov3/yolov3_tiny.hpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file methods/ann/models/yolov3_tiny.hpp
+ * @author Andrew Furey
+ *
+ * Definition of the YOLOv3-tiny model.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_MODELS_YOLOV3_TINY_HPP
+#define MLPACK_METHODS_ANN_MODELS_YOLOV3_TINY_HPP
+
+#include <mlpack/prereqs.hpp>
+
+#include <mlpack/methods/ann/dag_network.hpp>
+#include <mlpack/methods/ann/loss_functions/loss_functions.hpp>
+#include <mlpack/methods/ann/init_rules/init_rules.hpp>
+
+#include <mlpack/methods/ann/layer/layer_types.hpp>
+#include <mlpack/methods/ann/models/yolov3/yolov3_layer.hpp>
+
+namespace mlpack {
+/**
+ * YOLOv3-tiny is a small one-stage object detection model.
+ *
+ * For more information, refer to the following paper:
+ *
+ * @code
+ * @article{yolov3,
+ *   title     ={YOLOv3: An Incremental Improvement},
+ *   author    ={Redmon, Joseph and Farhadi, Ali},
+ *   journal   = {arXiv},
+ *   year      ={2018}
+ * }
+ * @endcode
+ *
+ */
+template <typename OutputLayerType = EmptyLoss,
+          typename InitializationRuleType = RandomInitialization,
+          typename MatType = arma::mat>
+class YOLOv3tiny
+{
+ public:
+
+  // Helper types.
+  using ModelType =
+    DAGNetwork<OutputLayerType, InitializationRuleType, MatType>;
+  using Type = typename MatType::elem_type;
+  using CubeType = typename GetCubeType<MatType>::type;
+
+  YOLOv3tiny() { /* Nothing to do. */ }
```

Comment:
So right now, it looks like the only way to actually use the YOLO model here is to train it manually.  Of course that is one way people would use these models, but what are you thinking for how users load pre-existing trained models?  I guess the typical "mlpack-like" way would be something like:

```c++
YOLOv3tiny y;
data::Load("model.bin", y);
// now you can predict or whatever
```

but for that we'd need a `serialize()` function.  I think it shouldn't be too hard to add?

We will also need to find a way to make a pretrained model available: for now we can just throw something on models.mlpack.org/ and link to it in our documentation.  Maybe later we can be a little smarter about it and allow loading over HTTP/HTTPS or something like that, but no need to do that in this PR.